src/util/text-utils.cc

```diff
@@ -208,47 +210,38 @@ bool convert_special_number(const std::string &str, T *out) {
   return false;
 }
 
+template <typename T>
 bool ConvertStringToReal(const std::string &str,
-                         double *out) {
-  const char *this_str = str.c_str();
-  char *end = NULL;
-  errno = 0;
-
-#if defined(_MSC_VER)
-  // TODO: check if the new MSVC already supports it
-  // depending on claims of the C++11 support, it should have
-  if (convert_special_number(str, out))
-    return true;
-#endif  // defined(_MSC_VER)
-
-  double d = KALDI_STRTOD(this_str, &end);
-  if (end != this_str)
-    while (isspace(*end)) end++;
-  if (end == this_str || *end != '\0' || errno != 0)
+                         T *out) {
+  // remove initial and final spaces
+  std::string trimmed_str = str;
+  Trim(&trimmed_str);
+
+  // if trimmed_str has more than one token,
+  // we will not try to convert it.
+  if (!IsToken(trimmed_str)) {
     return false;
-  *out = d;
-  return true;
-}
+  }
 
-bool ConvertStringToReal(const std::string &str,
-                         float *out) {
-  const char *this_str = str.c_str();
-  char *end = NULL;
-  errno = 0;
-
-#ifdef _MSC_VER
-  // TODO: check if the new MSVC already supports it
-  // depending on claims of the C++11 support, it should have
-  if (convert_special_number(str, out))
+  // deals with nan and inf
```

Comment:
If I understand this correctly, it does the nan stuff first, which is quite inefficient as it's not the normal case and it involves constructing maps, etc.   This function is called quite often.
Please make sure that the 'normal case' of this function, where it contains an actual number, is as efficient as possible.  So try to have it so that you just construct an istringstream, input it via >> to a float or double, and check its statuses and then return, as the normal code path.  All that stuff about Trim() should be unnecessary in the principal code path of the function.  Reading via >> to a float should probably skip over spaces by itself; if not, use >> std::ws or something like that.